frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+  expand_button.innerHTML = "Expand All"
+  expand_button.className = "expand"
+  expand_button.addEventListener('click', function () {
+    if ($(this).html() == 'Expand All') {
+      $(this).html('Collapse All');
+      $('.reportCollapse').collapse('show');
+    }
+    else {
+      $(this).html('Expand All');
+      $('.reportCollapse').collapse('hide');
+    }
+  })
+  accordion.appendChild(expand_button);
+
+  // global edit data type button
+  editButton = document.createElement('button');
+  editButton.setAttribute('id', 'editButton');
+  editButton.className = 'expand right-align';
+  editButton.innerHTML = 'Edit Global Data Type';
+  editButton.addEventListener('click', function () {
+    createEditDataTypeTable();
+    $('#globalDataTypeModal').modal();
+  });
+  accordion.appendChild(editButton);
+
+  var reportUl = document.createElement('ul');
+  reportUl.setAttribute('id', 'reportUl');
+  z = 0;
+
+  for (var x = 0; x < src_table_num; x++) {
+    initialPkSeqId[x] = new Array();
+    initialColNameArray[x] = new Array();
+    constraintTabCell[x] = new Array();
+    primaryTabCell[x] = new Array();
+    notPrimary[x] = new Array();
+    notNullFoundFlag[x] = new Array();
+    keyColumnMap[x] = new Array();
+    pkArray[x] = new Array();
+    selectedConstraints[x] = new Array();
+    spPlaceholder[x] = new Array();
+    count_sp[x] = new Array();
+    count_src[x] = new Array();
+    pksSp[x] = new Array();
+  }
+
+  interleaveApiCallResp = JSON.parse(localStorage.getItem('interleaveInfo'));
+
+  // creating table accordion one by one for each table
+  for (i = 0; i < src_table_num; i++) {
+    src_table = schemaConversionObj.SrcSchema[Object.keys(schemaConversionObj.ToSpanner)[i]];
+    src_table_name[i] = Object.keys(schemaConversionObj.ToSpanner)[i];
+    src_table_cols = src_table.ColNames;
+    sp_table = schemaConversionObj.SpSchema[src_table_name[i]];
+    sp_table_cols = sp_table.ColNames;
+    pkArray[i] = schemaConversionObj.SpSchema[Object.keys(schemaConversionObj.SpSchema)[i]].Pks;
+    pkSeqId[i] = 1;
+    for (var x = 0; x < pkArray[i].length; x++) {
+      pkArray[i][x].seqId = pkSeqId[i];
+      pkSeqId[i]++;
+    }
+    schemaConversionObj.SpSchema[src_table_name[i]].Pks = pkArray[i];
+
+    li = document.createElement('area');
+    reportUl.appendChild(li);
+
+    tableCard = document.createElement("div");
+    tableCard.className = "card";
+    tableCard.setAttribute('id', i);
+
+    tableCardHeader = document.createElement("div");
+    tableCardHeader.className = 'card-header report-card-header borderBottom' + panelBorderClass(conversionRateResp[src_table_name[i]]);
+    tableCardHeader.role = 'tab';
+
+    tableCardHeading = document.createElement("h5");
+    tableCardHeading.className = 'mb-0';
+
+    tableCardHeaderLink = document.createElement("a");
+    tableCardHeaderLink.innerHTML = `Table: ${Object.keys(schemaConversionObj.SrcSchema)[i]} <i class="fas fa-angle-down rotate-icon"></i>`
```

Comment:
XML tag has empty body, better to use like: `<i class="fas fa-angle-down rotate-icon"/>`

Reply:
code changes done accordingly